### ISSUE TITLE
XVLA Model Causes Stuttering Motion (Discontinuous Action Output) When Controlling AgileX Piper Robotic Arm

### ISSUE THREAD

**Issue by guanchzh** [huggingface/lerobot#2942]
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
- LeRobot version: 0.4.3
- Platform: Linux-5.15.0-139-generic-x86_64-with-glibc2.31
- Python version: 3.10.19
- Huggingface Hub version: 0.35.3
- Datasets version: 4.1.1
- Numpy version: 2.2.6
- FFmpeg version: 7.1.1
- PyTorch version: 2.10.0+cu130
- Is PyTorch built with CUDA support?: True
- Cuda version: 13.0
- GPU model: NVIDIA GeForce RTX 5060 Ti
- Using GPU in script?: <Yes>
- lerobot scripts: ['lerobot-calibrate', 'lerobot-dataset-viz', 'lerobot-edit-dataset', 'lerobot-eval', 'lerobot-find-cameras', 'lerobot-find-joint-limits', 'lerobot-find-port', 'lerobot-imgtransform-viz', 'lerobot-info', 'lerobot-record', 'lerobot-replay', 'lerobot-setup-motors', 'lerobot-teleoperate', 'lerobot-train']
```

### Description

- Bugs: When using the XVLA model to control the Piper robotic arm, the arm's movement exhibits noticeable stuttering (i.e., moving intermittently with pauses). Visualization reveals that the model's output action values change step-wise over time, resulting in discontinuous motion. The same self-collected dataset was used to train an ACT model, which runs smoothly and successfully completes the target tasks. 

- Questions:  What could be the potential reasons for this issue? Are there any debugging suggestions or known solutions?
- Features: The action visualization is shown in the figure below.

<img width="1395" height="777" alt="Image" src="https://github.com/user-attachments/assets/f6400557-9c91-40d2-adb2-d6a96ff52373" />

### Context & Reproduction

Below is the config.json of the xvla model.
{
    "type": "xvla",
    "n_obs_steps": 1,
    "input_features": {
        "observation.images.image": {
            "type": "VISUAL",
            "shape": [
                3,
                256,
                256
            ]
        },
        "observation.images.image2": {
            "type": "VISUAL",
            "shape": [
                3,
                256,
                256
            ]
        },
        "observation.state": {
            "type": "STATE",
            "shape": [
                8
            ]
        },
        "observation.images.empty_camera_0": {
            "type": "VISUAL",
            "shape": [
                3,
                224,
                224
            ]
        }
    },
    "output_features": {
        "action": {
            "type": "ACTION",
            "shape": [
                20
            ]
        }
    },
    "device": "cuda",
    "use_amp": false,
    "push_to_hub": true,
    "repo_id": null,
    "private": null,
    "tags": null,
    "license": null,
    "pretrained_path": null,
    "chunk_size": 32,
    "n_action_steps": 32,
    "normalization_mapping": {
        "STATE": "IDENTITY",
        "ACTION": "MEAN_STD",
        "VISUAL": "IDENTITY"
    },
    "florence_config": {
        "model_type": "florence2",
        "bos_token_id": 0,
        "eos_token_id": 2,
        "ignore_index": -100,
        "pad_token_id": 1,
        "projection_dim": 1024,
        "text_config": {
            "vocab_size": 51289,
            "activation_dropout": 0.1,
            "activation_function": "gelu",
            "add_bias_logits": false,
            "add_final_layer_norm": false,
            "attention_dropout": 0.1,
            "bos_token_id": 0,
            "classif_dropout": 0.1,
            "classifier_dropout": 0.0,
            "d_model": 1024,
            "decoder_attention_heads": 16,
            "decoder_ffn_dim": 4096,
            "decoder_layerdrop": 0.0,
            "decoder_layers": 12,
            "decoder_start_token_id": 2,
            "dropout": 0.1,
            "early_stopping": true,
            "encoder_attention_heads": 16,
            "encoder_ffn_dim": 4096,
            "encoder_layerdrop": 0.0,
            "encoder_layers": 12,
            "eos_token_id": 2,
            "forced_eos_token_id": 2,
            "forced_bos_token_id": 0,
            "gradient_checkpointing": false,
            "init_std": 0.02,
            "is_encoder_decoder": true,
            "label2id": {
                "LABEL_0": 0,
                "LABEL_1": 1,
                "LABEL_2": 2
            },
            "max_position_embeddings": 4096,
            "no_repeat_ngram_size": 3,
            "normalize_before": false,
            "num_hidden_layers": 12,
            "pad_token_id": 1,
            "scale_embedding": false,
            "num_beams": 3
        },
        "vision_config": {
            "model_type": "davit",
            "drop_path_rate": 0.1,
            "patch_size": [
                7,
                3,
                3,
                3
            ],
            "patch_stride": [
                4,
                2,
                2,
                2
            ],
            "patch_padding": [
                3,
                1,
                1,
                1
            ],
            "patch_prenorm": [
                false,
                true,
                true,
                true
            ],
            "enable_checkpoint": false,
            "dim_embed": [
                256,
                512,
                1024,
                2048
            ],
            "num_heads": [
                8,
                16,
                32,
                64
            ],
            "num_groups": [
                8,
                16,
                32,
                64
            ],
            "depths": [
                1,
                1,
                9,
                1
            ],
            "window_size": 12,
            "projection_dim": 1024,
            "visual_temporal_embedding": {
                "type": "COSINE",
                "max_temporal_embeddings": 100
            },
            "image_pos_embed": {
                "type": "learned_abs_2d",
                "max_pos_embeddings": 50
            },
            "image_feature_source": [
                "spatial_avg_pool",
                "temporal_avg_pool"
            ]
        },
        "vocab_size": 51289,
        "torch_dtype": "float32",
        "is_encoder_decoder": true
    },
    "tokenizer_name": "facebook/bart-large",
    "tokenizer_max_length": 1024,
    "tokenizer_padding_side": "right",
    "pad_language_to": "max_length",
    "hidden_size": 1024,
    "depth": 24,
    "num_heads": 16,
    "mlp_ratio": 4.0,
    "num_domains": 30,
    "len_soft_prompts": 32,
    "dim_time": 32,
    "max_len_seq": 512,
    "use_hetero_proj": false,
    "action_mode": "ee6d",
    "num_denoising_steps": 10,
    "use_proprio": true,
    "max_state_dim": 20,
    "domain_feature_key": null,
    "resize_imgs_with_padding": [
        224,
        224
    ],
    "num_image_views": 3,
    "empty_cameras": 1,
    "optimizer_lr": 0.0001,
    "optimizer_betas": [
        0.9,
        0.95
    ],
    "optimizer_eps": 1e-08,
    "optimizer_weight_decay": 0.0001,
    "optimizer_grad_clip_norm": 10.0,
    "scheduler_warmup_steps": 1000,
    "scheduler_decay_steps": 60000,
    "scheduler_decay_lr": 2.5e-06
}

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [ ] I am using the latest version of the `main` branch.
- [ ] I have verified this is not an environment-specific problem.

**Comment by pkooij** [comment]
@jadechoghari

**Comment by ColinMacM** [comment]
Also experienced this issue. Very annoying bug. Is there any VLA models that work with PiperX?
